src/main/scala/io/qbeast/spark/delta/SparkDeltaMetadataManager.scala

```diff
@@ -15,16 +16,29 @@ import org.apache.spark.sql.SparkSession
 /**
  * Spark+Delta implementation of the MetadataManager interface
  */
-object SparkDeltaMetadataManager extends MetadataManager[StructType, FileAction] {
+object SparkDeltaMetadataManager extends MetadataManager[StructType, FileAction, QbeastOptions] {
 
-  override def updateWithTransaction(tableID: QTableID, schema: StructType, append: Boolean)(
-      writer: => (TableChanges, IISeq[FileAction])): Unit = {
+  override def updateWithTransaction(
+      tableID: QTableID,
+      schema: StructType,
+      options: QbeastOptions,
+      append: Boolean)(writer: => (TableChanges, IISeq[FileAction])): Unit = {
 
     val deltaLog = loadDeltaQbeastLog(tableID).deltaLog
     val mode = if (append) SaveMode.Append else SaveMode.Overwrite
-    val options =
-      new DeltaOptions(Map("path" -> tableID.id), SparkSession.active.sessionState.conf)
-    val metadataWriter = DeltaMetadataWriter(tableID, mode, deltaLog, options, schema)
+    val conf = SparkSession.active.sessionState.conf
+    val deltaOptions = Map.newBuilder[String, String]
+    deltaOptions += "path" -> tableID.id
+    for (txnAppId <- options.txnAppId; txnVersion <- options.txnVersion) {
```

Comment:
Could there be more than one `txnAppId` and `txnVersion` set in the options?

Reply:
No, there cannot be more than one. I guess you are confused with `for` here. I borrowed this approach from Delta code where they handle the same options. The idea behind is that `for` allows to iterate over the Cartesian product of the specified generators. It is convenient because the syntax automatically unpack the elements, they are available under the variables declared tho the left of arrow. So the code in the body only works if and only if all the generators are not empty, and this is exactly what I need. An imperative analogue could be like the following:

```scala
if (options.txnAppId.notEmpty && options.txnVersion.notEmpty) {
  ...
}
```

Or if you like lambdas then

```scala
options.txnAppId.foreach(txnAppId => options.txnVersion.foreach(txnVersion => ...))
```

IMHO too many variants to my taste.